pkg/operator/ceph/csi/spec.go

```diff
@@ -802,8 +802,11 @@ func (r *ReconcileCSI) configureHolder(driver driverDetails, c ClusterDetail, tp
 		return errors.Wrapf(err, "failed to load ceph %q plugin holder template", driver.fullName)
 	}
 
-	// DO NOT set owner reference on ceph plugin holder daemonset, this DS must never restart unless
-	// the entire node is rebooted
```

Comment:
as having the ceph cluster as the owner it will not restart the DS.

Reply:
@BlaineEXE This comment was likely referring to not setting the ownerRef to the operator deployment, right? Setting the cluster CR as the ownerRef makes sense to me though.

Reply:
I don't think so. If there are multiple CephClusters, what happens when one of them is deleted? To my logic, the CSI drivers in general shouldn't have any owner set to a CephCluster because they are operator-/cluster-scoped. We don't want to accidentally delete the CSI driver when one of many CephClusters are removed.

Reply:
Oh of course, the holder daemonset has the same scope as the csi driver, which means it may be serving multiple clusters. So the question is what (if any) resource can the holder set as its owner reference so that it will be cleaned up during uninstall, but only after all cephcluster resources.

Reply:
@BlaineEXE @travisn These are the special daemonset holder as you know and they get created in the ceph cluster namespace not on the operator namespace like cephcsi plugin daemonset. they are unique per ceph cluster as multiple ceph clusters can be created with different multus networks.

Reply:
Ok, now I see where the holders are created per cluster. @BlaineEXE Does that answer your concern as well?

Reply:
Yes, I forgot that detail

Reply:
Actually, there is still an issue with this... Even though a separate daemonset is created for each cephcluster, they are still created in the operator namespace. The holder daemonsets are named `csi-rbdplugin-holder-<cluster>` so they can all be created in the same namespace with the operator. But in this case, the owner references cannot be set on the cephcluster in a different namespace, or else it results in this error:
```
2023-08-21 16:39:38.454427 E | ceph-csi: failed to reconcile failed to configure ceph csi: failed to start ceph csi drivers: failed to configure holder: failed to configure holder "rbd" for "my-cluster"/"other-ceph": failed to set owner reference to plugin holder "rook-ceph.rbd.csi.ceph.com": cross-namespace owner references are disallowed, owner's namespace other-ceph, obj's namespace rook-ceph
```

It may be a breaking change at this point, but can the holder daemonsets be created in the same namespace as the cephcluster? That would be necessary for setting the ownerReference like this, except that the CSI driver likely requires the holder pod to be in the same namespace as itself.

Reply:
For a scoped fix, we could add the ownerReference to the holder daemonset only if is in the same namespace as the operator. This should cover the common scenario anyway.

Reply:
@leseb any idea why we choose to create a holder daemonset set in the operator namespace not in the cluster namespace?